### PR TITLE
Config: Converts two LUT maps over linear scan arrays

### PR DESCRIPTION
**Pull request by Sonicadvance1**
These two maps used for environment lookup translations were getting
globally initialized and then registers with atexit handlers.

Switch over to a constexpr array and just do linear scans. This plus
short-circuiting the environment loader so it skips all entries that
don't start with `FEX_` has the side benefit of cutting the CPU time to
1/10th the time.

This plus https://github.com/FEX-Emu/FEX/pull/3917 removes the global static initializers entirely from
this file.

### CODE REVIEW

**Comment by alyssarosenzweig** [comment]
I've never heard of this library. How will this affect #2996?

**Comment by Sonicadvance1** [comment]
> I've never heard of this library. How will this affect #2996?

It's a header only library so it shouldn't cause issues. If they have problems with it then we'll need to find another solution for them, since there is no functional equivalent inside of C/C++.

**Comment by neobrain** [comment]
Why aren't you converting the type to `std::pair<const char*, FEXCore::Config::ConfigOption>>[]` instead of pulling in a new library? The code isn't performance critical as far as I can see, so a theoretical degradation from O(1) to O(log n) shouldn't matter. Since `n` is small here, flat memory may actually perform better.

As an aside, since these PRs keep coming up, could we document the rationale for avoiding global static initializers somewhere?

**Comment by Sonicadvance1** [comment]
Dropped the constexpr maps and switched to linear scan of an array. Also short-circuited the logic in the environment config loader so we stop loading all environment variables in to the temporary map, which accelerates execution time to 1/10th of what it was before.

File: Source/Common/CMakeLists.txt [hunk 1]
```diff
@@ -17,4 +17,5 @@ endif()
 add_library(${NAME} STATIC ${SRCS})
 target_link_libraries(${NAME} FEXCore_Base cpp-optparse tiny-json json-maker FEXHeaderUtils)
 target_include_directories(${NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/External/cpp-optparse/)
+target_include_directories(${NAME} PRIVATE ${CMAKE_SOURCE_DIR}/External/eternal/include/)
```
Comment (alyssarosenzweig) [inline Source/Common/CMakeLists.txt:20]: please drop this

Comment (Sonicadvance1) [reply Source/Common/CMakeLists.txt]: Oops, dropped.

**Review by alyssarosenzweig — APPROVED** [review]

**Pull request merged**